ledger/src/advance.rs

```diff
@@ -19,35 +19,181 @@ impl<N: Network, C: ConsensusStorage<N>> Ledger<N, C> {
     /// Returns a candidate for the next block in the ledger, using a committed subdag and its transmissions.
     pub fn prepare_advance_to_next_quorum_block<R: Rng + CryptoRng>(
         &self,
-        subdag: Subdag<N>,
-        transmissions: IndexMap<TransmissionID<N>, Transmission<N>>,
+        mut subdag: Subdag<N>,
+        subdag_transmissions: SubdagTransmissions<N>,
```

Comment:
This function is hella ugly now but we can create a new issue for cleaning it up